libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Testing/TestActions/SetProperties.cs

```diff
@@ -61,7 +61,7 @@ public async override Task ExecuteAsync(TestAdapter adapter, BotCallbackHandler
             }
             else
             {
-                throw new Exception("No inspector to use for setting properties");
+                throw new InvalidOperationException("No inspector to use for setting properties");
```

Comment:
This should be an `ArgumentException`.

Reply:
we decided to use InvalidOperation here because inspector is null by default, so this can be thrown even if the inspector method is not being used as its optional

Reply:
I don't see the difference, but it's ok. If you consider `InvalidOperationException` is more appropriate 👍.

Reply:
Truth be told neither is 100% correct because the code contradicts itself, but we chose _InvalidOperation_ over _Argument_ because the argument being null is supposed to be OK, at least the code states it that way.